workspaces/backend/api/workspacekinds_handler.go

```diff
@@ -154,8 +154,24 @@ func (a *App) GetWorkspaceKindsHandler(w http.ResponseWriter, r *http.Request, _
 //	@Router			/workspacekinds [post]
 func (a *App) CreateWorkspaceKindHandler(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
 
+	// Parse dry_run query parameter using helper
+	isDryRun, err := a.GetBooleanQueryParameter(w, r, "dry_run")
+	if err != nil {
+		return
+	}
+	hasDryRun := r.URL.Query().Get("dry_run") != ""
+
 	// validate the Content-Type header
-	if success := a.ValidateContentType(w, r, MediaTypeYaml); !success {
+	if !a.ValidateContentType(w, r, MediaTypeYaml) {
+		if hasDryRun {
+			// Special case: dry_run provided with wrong media type → 400
+			a.badRequestResponse(w, r,
+				fmt.Errorf("dry_run is only supported when Content-Type: %s", MediaTypeYaml))
+		} else {
+			// Normal case: wrong media type → 415
+			a.unsupportedMediaTypeResponse(w, r,
+				fmt.Errorf("only %s is supported", MediaTypeYaml))
+		}
```

Comment:
I realize this implementation is specifically following the GH issue guidance... but now that I am re-reviewing things (and note: i wrote that issue!) - I can't for the life of me think why this would be desirable!

We shouldn't carve out a specific error message here for `dry_run` parameter... if the `ValidateContentType` check fails... we should just be doing
```suggestion
	if !a.ValidateContentType(w, r, MediaTypeYaml) {
		a.unsupportedMediaTypeResponse(w, r,
		  fmt.Errorf("only %s is supported", MediaTypeYaml))
```

This also means we don't need this additional "look up":
```
hasDryRun := r.URL.Query().Get("dry_run") != ""
```

which is a nice added benefit - as the `URL.Query.Get(...)` code now can purely be contained within the helper function.

sorry for that poor initial guidance!